test/jest/Bladeburner/Actions.test.ts

```diff
@@ -0,0 +1,358 @@
+import { Bladeburner } from "../../../src/Bladeburner/Bladeburner";
+import { PlayerObject } from "../../../src/PersonObjects/Player/PlayerObject";
+import { Player, setPlayer } from "@player";
+import { BlackOperation, Contract, GeneralAction, Operation } from "../../../src/Bladeburner/Actions";
+import {
+  BladeburnerActionType,
+  BladeburnerContractName,
+  BladeburnerGeneralActionName,
+  BladeburnerOperationName,
+  BladeburnerSkillName,
+  CityName,
+  CrimeType,
+} from "@enums";
+import { FormatsNeedToChange } from "../../../src/ui/formatNumber";
+import { CrimeWork } from "../../../src/Work/CrimeWork";
+import type { Action, ActionIdentifier } from "../../../src/Bladeburner/Types";
+import type { Skills } from "@nsdefs";
+import { BlackOperations } from "../../../src/Bladeburner/data/BlackOperations";
+
+describe("Bladeburner Actions", () => {
+  const SampleContract = Contract.createId(BladeburnerContractName.Tracking);
+  const SampleGeneralAction = GeneralAction.createId(BladeburnerGeneralActionName.Diplomacy);
+  const SampleOperation = Operation.createId(BladeburnerOperationName.Assassination);
+  const SampleBlackOp = BlackOperations["Operation Centurion"].id;
+  const ENOUGH_TIME_TO_FINISH_ACTION = 1e5;
+
+  /** Why 1e14? The stat tests need to be able to detect an EXP change of cca. 0.01
+   * 1e14 is the last threshold where 1e14 + 0.01 > 1e14
+   *
+   * Bug? Coincidentally this also means that stat gains stop at that level unless
+   * exp multipliers play a significant role
+   *
+   * Example: 1e14 strength exp gives ~~831 strength skill without augs/modifiers
+   */
+  const HIGH_STAT_EXP = 1e14;
+  const LOW_STAT_EXP = 1e6;
+
+  let inst: Bladeburner;
+
+  const instanceUsedForTestGeneration = new Bladeburner();
+  const CITIES = <CityName[]>Object.keys(instanceUsedForTestGeneration.cities);
+
+  /** All the tests depend on this assumption */
+  it("always succeeds with optimal stats, rank, stamina and city chaos levels", () => {
```

Comment:
This is not a test directly, but rather a shared assumption that breaks all reliability should this one not pass.

Expect this to change only if we significantly alter balance in favor of making BB harder on combat stats, which I deemed unlikely.